Author: Nilesh Patra <npatra974@gmail.com>
        xiao sheng wen <atzlinux@sina.com>
Description: Use external libfmt by default
Last-Changed: Fri, 10 Jun 2022
Forwarded: not-needed
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,7 +88,7 @@ option(SPDLOG_BUILD_WARNINGS "Enable com
 option(SPDLOG_SYSTEM_INCLUDES "Include as system headers (skip for clang-tidy)." OFF)
 option(SPDLOG_INSTALL "Generate the install target" ${SPDLOG_MASTER_PROJECT})
 option(SPDLOG_USE_STD_FORMAT "Use std::format instead of fmt library." OFF)
-option(SPDLOG_FMT_EXTERNAL "Use external fmt library instead of bundled" OFF)
+option(SPDLOG_FMT_EXTERNAL "Use external fmt library instead of bundled" ON)
 option(SPDLOG_FMT_EXTERNAL_HO "Use external fmt header-only library instead of bundled" OFF)
 option(SPDLOG_NO_EXCEPTIONS "Compile with -fno-exceptions. Call abort() on any spdlog exceptions" OFF)
 
--- a/include/spdlog/fmt/bin_to_hex.h
+++ b/include/spdlog/fmt/bin_to_hex.h
@@ -5,6 +5,7 @@
 
 #pragma once
 
+#include <spdlog/tweakme.h>
 #include <cctype>
 #include <spdlog/common.h>
 
--- a/include/spdlog/fmt/fmt.h
+++ b/include/spdlog/fmt/fmt.h
@@ -4,7 +4,7 @@
 //
 
 #pragma once
-
+#include <spdlog/tweakme.h>
 //
 // Include a bundled header-only copy of fmtlib or an external one.
 // By default spdlog include its own copy.
--- a/include/spdlog/fmt/ostr.h
+++ b/include/spdlog/fmt/ostr.h
@@ -8,6 +8,7 @@
 // include bundled or external copy of fmtlib's ostream support
 //
 
+#include <spdlog/tweakme.h>
 #if !defined(SPDLOG_USE_STD_FORMAT)
 #    if !defined(SPDLOG_FMT_EXTERNAL)
 #        ifdef SPDLOG_HEADER_ONLY
--- a/include/spdlog/tweakme.h
+++ b/include/spdlog/tweakme.h
@@ -78,7 +78,7 @@
 // In this case spdlog will try to include <fmt/format.h> so set your -I flag
 // accordingly.
 //
-// #define SPDLOG_FMT_EXTERNAL
+#define SPDLOG_FMT_EXTERNAL 1
 ///////////////////////////////////////////////////////////////////////////////
 
 ///////////////////////////////////////////////////////////////////////////////
